### PR TITLE
"role attribute has valid value" (674b10): Prepare for TF review

### PR DESCRIPTION
**Pull request by Jym77**
Preparing the rule for TF review.
Mostly upgrading the example descriptions and killing notes.

Closes issue(s):

- N/A

Need for Final Call:
This will require a 1 week Final Call 

---

### **When creating PR:**

- [X] Make sure you're requesting to **pull a branch** (right side) to the `develop` branch (left side).

### **After creating PR:**

- [X] Add yourself (and co-authors) as "Assignees" for PR.
- [X] Add label to indicate if it's a `Rule`, `Definition` or `Chore`.
- [ ] [Link the PR to any issue it solves](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue). This will be done automatically by [referencing the issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) at the top of this comment in the indicated place.
- [X] Optionally request feedback from anyone in particular by assigning them as "Reviewers".

### **When merging a PR:**

- [ ] Close any issue that the PR resolves. This will happen automatically upon merging if the PR was correctly linked to the issue, e.g. by referencing the issue at the top of this comment.

## How to Review And Approve

- Go to the “Files changed” tab
- Here you will have the option to leave comments on different lines.
- Once the review is completed, find the “Review changes” button in the top right, select “Approve” (if you are really confident in the rule) or "Request changes" and click “Submit review”.
- Make sure to also review the proposed Final Call period. In case of disagreement, the longer period wins.

### CODE REVIEW

File: _rules/role-attribute-valid-value-674b10.md [hunk 1]
```diff
@@ -60,23 +65,23 @@ Older browsers do not support more than one token in the value for a role attrib
...
 <span role="button link"></span>
 ```
 
 #### Passed Example 3
 
-Element with at least one valid `role` value.
+This [`role` attribute][role attribute] contains two tokens, and one of these tokens is a valid [WAI-ARIA role][].
 
 ```html
 <img role="img xyz" src="/test-assets/shared/w3c-logo.png" alt="W3C logo" />
```
Comment (Jym77) [inline _rules/role-attribute-valid-value-674b10.md:87]: This example could be upgraded a bit by using another value than "xyz" that looks like a "plausible typo".
I preferred not touching the examples to keep implementation data.

File: _rules/role-attribute-valid-value-674b10.md [hunk 2]
```diff
@@ -86,23 +91,23 @@ Element with at least one valid `role` value.
 
 #### Failed Example 1
 
-Element with invalid `role` value.
+This [`role` attribute][role attribute] contains one token, but this token is not a valid role in any of the [WAI-ARIA specifications][].
 
 ```html
 <input role="invalid" value="123" />
```
Comment (Jym77) [inline _rules/role-attribute-valid-value-674b10.md:97]: Same. A more plausible failing value could be used.

File: _rules/role-attribute-valid-value-674b10.md [hunk 3]
```diff
@@ -86,23 +91,23 @@ Element with at least one valid `role` value.
...
 <input type="text" role="invalid role" />
 ```
 
 #### Failed Example 3
 
-Element with role attribute that is not empty (""), neither a valid `role` value.
+This [`role` attribute][role attribute] contains one token, but this token is not a valid role in any of the [WAI-ARIA specifications][].
 
 ```html
 <input type="text" role="#" />
```
Comment (Jym77) [inline _rules/role-attribute-valid-value-674b10.md:113]: I am not sure this example is really needed as it is essentially the same problem as Failed Example 1.
Should we remove it?

File: _rules/role-attribute-valid-value-674b10.md [hunk 4]
```diff
@@ -112,44 +117,48 @@ Element with role attribute that is not empty (""), neither a valid `role` value
```
Comment (Jym77) [inline _rules/role-attribute-valid-value-674b10.md:117]: Inapplicable examples have been moved around to mimic the order of conditions in the applicability.
They haven't change, despite what Github believes 😉

File: _rules/role-attribute-valid-value-674b10.md [hunk 5]
```diff
@@ -30,26 +30,31 @@ acknowledgments:
 
 ## Applicability
 
-Any `role` attribute that is neither empty ("") nor only [ASCII whitespace][], and that is specified on an HTML or SVG element that is [included in the accessibility tree][].
+Any [`role` attribute][role attribute] for which all the following are true:
```
Comment (WilcoFiers) [inline _rules/role-attribute-valid-value-674b10.md:33]: Linking attributes is only necessary if they are ambiguous. There is only one type of role attribute in HTML.
```suggestion
Any `role` attribute for which all the following are true:
```

File: _rules/role-attribute-valid-value-674b10.md [hunk 6]
```diff
@@ -30,26 +30,31 @@ acknowledgments:
...
 
 ## Expectation
 
-Each test target has at least one token which is a valid value corresponding to a non-abstract role from [WAI-ARIA Specifications](#wai-aria-specifications).
+Each test target has at least one token which is a valid value corresponding to a non-abstract role from [WAI-ARIA Specifications][].
 
 ## Assumptions
 
-- The ARIA `role` is being used to comply to WCAG.
+The ARIA `role` is being used to comply to WCAG. If compliance to WCAG is done in other ways, this rule may fail while [Success Criterion 4.1.2 Name, Role, Value][sc412] is still satisfied.
```
Comment (WilcoFiers) [inline _rules/role-attribute-valid-value-674b10.md:44]: I'm not sure this made the assumption any clearer.... Perhaps we can do:

```suggestion
If the `role` attribute is used for something other than giving elements an [explicit role][], the success criterion could be satisfied even if the rule failed. For example, if for some reason the `role` attribute contains a date, this rule will fail. The implicit role of such an element may mean be sufficient to satisfy the success criterion.```

File: _rules/role-attribute-valid-value-674b10.md [hunk 7]
```diff
@@ -60,23 +65,23 @@ Older browsers do not support more than one token in the value for a role attrib
 
 #### Passed Example 1
 
-Element with valid `role` value.
+This [`role` attribute][role attribute] contains one token, and this token is a valid [WAI-ARIA role][].
```
Comment (WilcoFiers) [inline _rules/role-attribute-valid-value-674b10.md:68]:
```suggestion
This `role` attribute contains one token, and this token is a valid [WAI-ARIA role][].
```

File: _rules/role-attribute-valid-value-674b10.md [hunk 8]
```diff
@@ -60,23 +65,23 @@ Older browsers do not support more than one token in the value for a role attrib
...
 
 ```html
 <input type="text" role="textbox" />
 ```
 
 #### Passed Example 2
 
-Element with multiple valid `role` values.
+This [`role` attribute][role attribute] contains two tokens, and these tokens are both valid [WAI-ARIA roles][wai-aria role].
```
Comment (WilcoFiers) [inline _rules/role-attribute-valid-value-674b10.md:76]:
```suggestion
This `role` attribute contains two tokens, and these tokens are both valid [WAI-ARIA roles][wai-aria role].
```

File: _rules/role-attribute-valid-value-674b10.md [hunk 9]
```diff
@@ -60,23 +65,23 @@ Older browsers do not support more than one token in the value for a role attrib
...
 
 ```html
 <span role="button link"></span>
 ```
 
 #### Passed Example 3
 
-Element with at least one valid `role` value.
+This [`role` attribute][role attribute] contains two tokens, and one of these tokens is a valid [WAI-ARIA role][].
```
Comment (WilcoFiers) [inline _rules/role-attribute-valid-value-674b10.md:84]:
```suggestion
This `role` attribute contains two tokens, and one of these tokens is a valid [WAI-ARIA role][].
```

File: _rules/role-attribute-valid-value-674b10.md (hunk 1 above)
Comment (WilcoFiers) [reply _rules/role-attribute-valid-value-674b10.md:87]: I like that idea. I think we should just make that change.

File: _rules/role-attribute-valid-value-674b10.md [hunk 10]
```diff
@@ -86,23 +91,23 @@ Element with at least one valid `role` value.
 
 #### Failed Example 1
 
-Element with invalid `role` value.
+This [`role` attribute][role attribute] contains one token, but this token is not a valid role in any of the [WAI-ARIA specifications][].
```
Comment (WilcoFiers) [inline _rules/role-attribute-valid-value-674b10.md:94]:
```suggestion
This `role` attribute contains one token, but this token is not a valid role in any of the [WAI-ARIA specifications][].
```

File: _rules/role-attribute-valid-value-674b10.md [hunk 11]
```diff
@@ -86,23 +91,23 @@ Element with at least one valid `role` value.
...
 
 ```html
 <input role="invalid" value="123" />
 ```
 
 #### Failed Example 2
 
-Element with multiple invalid `role` value.
+This [`role` attribute][role attribute] contains two tokens, but none of these tokens is a valid role in any of the [WAI-ARIA specifications][].
```
Comment (WilcoFiers) [inline _rules/role-attribute-valid-value-674b10.md:102]:
```suggestion
This `role` attribute contains two tokens, but none of these tokens is a valid role in any of the [WAI-ARIA specifications][].
```

File: _rules/role-attribute-valid-value-674b10.md [hunk 12]
```diff
@@ -86,23 +91,23 @@ Element with at least one valid `role` value.
...
 
 ```html
 <input type="text" role="invalid role" />
 ```
 
 #### Failed Example 3
 
-Element with role attribute that is not empty (""), neither a valid `role` value.
+This [`role` attribute][role attribute] contains one token, but this token is not a valid role in any of the [WAI-ARIA specifications][].
```
Comment (WilcoFiers) [inline _rules/role-attribute-valid-value-674b10.md:110]:
```suggestion
This `role` attribute contains one token, but this token is not a valid role in any of the [WAI-ARIA specifications][].
```

File: _rules/role-attribute-valid-value-674b10.md (hunk 3 above)
Comment (WilcoFiers) [reply _rules/role-attribute-valid-value-674b10.md:113]: Agreed, lets take it out.

File: _rules/role-attribute-valid-value-674b10.md [hunk 13]
```diff
@@ -112,44 +117,48 @@ Element with role attribute that is not empty (""), neither a valid `role` value
 
 #### Inapplicable Example 1
 
-Element with `role` attribute that is empty ("").
+There is no [`role` attribute][role attribute].
```
Comment (WilcoFiers) [inline _rules/role-attribute-valid-value-674b10.md:120]:
```suggestion
There is no `role` attribute.
```

File: _rules/role-attribute-valid-value-674b10.md [hunk 14]
```diff
@@ -112,44 +117,48 @@ Element with role attribute that is not empty (""), neither a valid `role` value
...
 
 ```html
-<div role="">Some Content</div>
+<div>Some Content</div>
 ```
 
 #### Inapplicable Example 2
 
-Element with role attribute that is only [ASCII whitespace][].
+This [`role` attribute][role attribute] has no value.
```
Comment (WilcoFiers) [inline _rules/role-attribute-valid-value-674b10.md:128]:
```suggestion
This `role` attribute has no value.
```

File: _rules/role-attribute-valid-value-674b10.md [hunk 15]
```diff
@@ -112,44 +117,48 @@ Element with role attribute that is not empty (""), neither a valid `role` value
...
 
 ```html
-<input type="text" role=" " />
+<div role>Some Content</div>
 ```
 
 #### Inapplicable Example 3
 
-Element does not have `role` attribute.
+This [`role` attribute][role attribute] is empty ("").
```
Comment (WilcoFiers) [inline _rules/role-attribute-valid-value-674b10.md:136]:
```suggestion
This `role` attribute is empty ("").
```

File: _rules/role-attribute-valid-value-674b10.md [hunk 16]
```diff
@@ -112,44 +117,48 @@ Element with role attribute that is not empty (""), neither a valid `role` value
...
 
 ```html
-<div>Some Content</div>
+<div role="">Some Content</div>
 ```
 
 #### Inapplicable Example 4
 
-Element with null `role` attribute.
+This [`role` attribute][role attribute] is only [ASCII whitespace][].
```
Comment (WilcoFiers) [inline _rules/role-attribute-valid-value-674b10.md:144]:
```suggestion
This `role` attribute is only [ASCII whitespace][].
```

File: _rules/role-attribute-valid-value-674b10.md [hunk 17]
```diff
@@ -112,44 +117,48 @@ Element with role attribute that is not empty (""), neither a valid `role` value
...
 
 ```html
-<div role>Some Content</div>
+<input type="text" role=" " />
 ```
 
 #### Inapplicable Example 5
 
-Element that is not [included in the accessibility tree][].
+This [`role` attribute][role attribute] is specified on an element which is not [included in the accessibility tree][].
```
Comment (WilcoFiers) [inline _rules/role-attribute-valid-value-674b10.md:152]:
```suggestion
This `role` attribute is specified on an element which is not [included in the accessibility tree][].
```

File: _rules/role-attribute-valid-value-674b10.md [hunk 18]
```diff
@@ -112,44 +117,48 @@ Element with role attribute that is not empty (""), neither a valid `role` value
...
 
 ```html
 <div aria-hidden="true" role="banner">Some Content</div>
 ```
 
 [ascii whitespace]: https://infra.spec.whatwg.org/#ascii-whitespace 'Definition of ASCII whitespace'
 [included in the accessibility tree]: #included-in-the-accessibility-tree 'Definition of included in the accessibility tree'
+[role attribute]: https://www.w3.org/TR/role-attribute/ 'Specification of the Role attribute'
```
Comment (WilcoFiers) [inline _rules/role-attribute-valid-value-674b10.md:160]:
```suggestion
```

**Comment by Jym77** [comment]
@WilcoFiers very good points. I realised the assumption needed a bit more oomph and ended up in a more general version of your suggestion.
Plus worked to improve the examples.

**Comment by Jym77** [comment]
Final Call ends on June 30.

**Comment by Jym77** [comment]
Final Call has ended. Merging.

**Pull request merged**